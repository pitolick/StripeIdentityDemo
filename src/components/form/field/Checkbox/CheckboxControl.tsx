/**
 * チェックボックス制御コンポーネント（react-hook-form）
 * @module CheckboxControl
 * @todo カードスタイルのときにTabでフォーカスした場合にアウトラインがでないのを修正
 */
import React, { forwardRef } from 'react'
import { Checkbox, CheckboxProps } from './Checkbox'
import { InputBox, InputBoxProps } from '@/components/form/parts'

/**
 * チェックボックスコンポーネントのプロパティ
 * @typedef {Object} CheckboxControlProps
 * @property {React.InputHTMLAttributes<HTMLInputElement>} - チェックボックスタグのHTML属性
 */

export interface CheckboxControlProps extends React.InputHTMLAttributes<HTMLInputElement>, Omit<InputBoxProps, 'children'> {
  /**
   * リスト
   */
  lists: CheckboxProps[]
  /**
   * ラベルのタイプ
   */
  labelType?: 'default'
  /**
   * 並び方
   */
  direction?: 'row' | 'col'
  /**
   * チェックボックスの位置
   */
  checkboxPosition?: 'center' | 'top' | 'hidden'
  /**
   * チェックボックスの色
   */
  checkboxColor?: 'black' | 'red' | 'blue' | 'gray'
}

/**
 * チェックボックス制御コンポーネント（react-hook-form）
 * @param {CheckboxControlProps} props - チェックボックスコンポーネントのプロパティ
 * @returns {JSX.Element} - チェックボックスコンポーネント
 */
export const CheckboxControl = forwardRef<HTMLInputElement, CheckboxControlProps>((props, ref) => {
  const {
    lists,
    nameId,
    label,
    note,
    required = undefined,
    error,
    labelType = 'default',
    direction = 'row',
    checkboxPosition = 'center',
    checkboxColor = 'blue',
    contentWidth,
    ...checkboxControlProps
  } = props

  const checkboxList = lists.map((list, index) => {
    const { labelContents, checkboxClassName, labelClassName, ...checkboxProps } = list
    // class名
    let checkboxClassNameProps = `${checkboxClassName ? checkboxClassName : ''}`
    let labelClassNameProps = `${labelClassName ? labelClassName : ''} inline-block relative pl-5`

    // チェックボックスの位置
    switch (checkboxPosition) {
      case 'center':
        checkboxClassNameProps = checkboxClassNameProps + ' absolute margin-auto top-1/2 transform -translate-y-1/2'
        break
      case 'top':
        checkboxClassNameProps = checkboxClassNameProps + ' absolute margin-auto'
        // ラベルのタイプによって分岐
        switch (labelType) {
          case 'default':
            checkboxClassNameProps = checkboxClassNameProps + ' top-1'
            break
        }
        break
      case 'hidden':
        checkboxClassNameProps = 'hidden'
        break
    }

    // エラー時のclass名
    let errorLabelClass = ''
    let errorCheckboxClass = ''

    // ラベルのタイプ
    switch (labelType) {
      case 'default':
        checkboxClassNameProps = checkboxClassNameProps + ' left-0'
        errorLabelClass = error ? ' text-red' : ''
        errorCheckboxClass = error ? ' bg-red-pale text-red border-red' : ''
        break
    }

    // チェックボックスの色
    switch (checkboxColor) {
      case 'black':
        checkboxClassNameProps = checkboxClassNameProps + ' checked:text-black focus:text-black'
        break
      case 'red':
        checkboxClassNameProps = checkboxClassNameProps + ' checked:text-red focus:text-red'
        break
      case 'blue':
        checkboxClassNameProps = checkboxClassNameProps + ' checked:text-blue focus:text-blue'
        break
      case 'gray':
        checkboxClassNameProps = checkboxClassNameProps + ' checked:text-gray-700 focus:text-gray-700'
        break
    }

    // カードコンポーネント
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
      return <>{children}</>
    }

    return (
      <div key={`${nameId}_${index}`} className={`${direction === 'col' && 'w-full'} ${direction === 'row' && contentWidth === 'full' && 'flex-auto'}`}>
        <Wrapper>
          <Checkbox
            ref={ref}
            labelContents={labelContents}
            checkboxClassName={`${checkboxClassNameProps} ${errorCheckboxClass}`}
            labelClassName={`${labelClassNameProps} ${errorLabelClass}`}
            wrapper={true}
            {...checkboxProps}
            {...checkboxControlProps}
          />
        </Wrapper>
      </div>
    )
  })

  return (
    <InputBox nameId={nameId} label={label} note={note} required={required} error={error} contentWidth={contentWidth}>
      {/* チェックボックス */}
      <div className={`${'flex-' + direction} ${contentWidth === 'full' && 'w-full'} flex flex-wrap gap-3`}>{checkboxList}</div>
    </InputBox>
  )
})

CheckboxControl.displayName = 'CheckboxControl'
