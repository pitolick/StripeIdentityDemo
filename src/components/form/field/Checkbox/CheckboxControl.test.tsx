import React from 'react'
import { render } from '@testing-library/react'
import { CheckboxControl, CheckboxControlProps } from './CheckboxControl'

describe('CheckboxControl', () => {
  const defaultProps: CheckboxControlProps = {
    lists: [
      {
        labelContents: 'オプション1',
        value: 'option1',
      },
      {
        labelContents: 'オプション2',
        value: 'option2',
      },
    ],
    nameId: 'checkboxField',
    label: 'チェックボックスフィールド',
  }

  it('正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('labelTypeプロパティがdefaultの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} labelType="default" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('labelTypeプロパティがdefaultでerrorがtrueの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} labelType="default" error={{ type: 'test' }} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('labelTypeプロパティがcardの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('labelTypeプロパティがcardでerrorがtrueの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} error={{ type: 'test' }} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('directionプロパティがrowの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} direction="row" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('directionプロパティがcolの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} direction="col" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxPositionプロパティがcenterの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxPosition="center" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxPositionプロパティがtopでlabelTypeプロパティがdefaultの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxPosition="top" labelType="default" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxPositionプロパティがtopでlabelTypeプロパティがcardの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxPosition="top" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxPositionプロパティがhiddenの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxPosition="hidden" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxColorプロパティがblackの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxColor="black" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxColorプロパティがredの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxColor="red" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxColorプロパティがblueの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxColor="blue" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('checkboxColorプロパティがgrayの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} checkboxColor="gray" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('listsにcheckboxClassName labelClassNameプロパティが指定されている場合、正しくレンダリングされること', () => {
    const { container } = render(
      <CheckboxControl
        {...defaultProps}
        lists={[
          {
            labelContents: 'オプション1',
            value: 'option1',
            checkboxClassName: 'custom-checkbox',
            labelClassName: 'custom-label',
          },
          {
            labelContents: 'オプション2',
            value: 'option2',
            checkboxClassName: 'custom-checkbox',
            labelClassName: 'custom-label',
          },
        ]}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('directionがrowでcontentWidthがfullの場合、正しくレンダリングされること', () => {
    const { container } = render(<CheckboxControl {...defaultProps} direction="row" contentWidth="full" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
