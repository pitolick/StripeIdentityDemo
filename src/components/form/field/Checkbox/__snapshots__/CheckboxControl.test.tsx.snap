// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxControl checkboxColorプロパティがblackの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-black focus:text-black "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-black focus:text-black "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxColorプロパティがblueの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxColorプロパティがgrayの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-gray-700 focus:text-gray-700 "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-gray-700 focus:text-gray-700 "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxColorプロパティがredの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-red focus:text-red "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-red focus:text-red "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxPositionプロパティがcenterの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxPositionプロパティがhiddenの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent hidden left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent hidden left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxPositionプロパティがtopでlabelTypeプロパティがcardの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl checkboxPositionプロパティがtopでlabelTypeプロパティがdefaultの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl directionがrowでcontentWidthがfullの場合、正しくレンダリングされること 1`] = `
<div
  class="w-full flex-auto space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row w-full flex flex-wrap gap-3"
      >
        <div
          class="false flex-auto"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false flex-auto"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl directionプロパティがcolの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-col false flex flex-wrap gap-3"
      >
        <div
          class="w-full false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="w-full false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl directionプロパティがrowの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl labelTypeプロパティがcardでerrorがtrueの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  text-red cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  text-red cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
    <p
      class="text-sm text-red"
    />
  </div>
</div>
`;

exports[`CheckboxControl labelTypeプロパティがcardの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl labelTypeプロパティがdefaultでerrorがtrueの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  text-red cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  text-red cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
    <p
      class="text-sm text-red"
    />
  </div>
</div>
`;

exports[`CheckboxControl labelTypeプロパティがdefaultの場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl listsにcheckboxClassName labelClassNameプロパティが指定されている場合、正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class="custom-label inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent custom-checkbox absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class="custom-label inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent custom-checkbox absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckboxControl 正しくレンダリングされること 1`] = `
<div
  class="false space-y-2 false"
>
  <div
    class="flex items-center justify-start gap-x-2"
  >
    <label
      class="block text-sm font-bold"
      for="checkboxField"
    >
      チェックボックスフィールド
    </label>
  </div>
  <div
    class="space-y-2 false"
  >
    <div
      class="flex w-full items-center justify-start gap-x-3"
    >
      <div
        class="flex-row false flex flex-wrap gap-3"
      >
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option1"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション1
            </span>
          </label>
        </div>
        <div
          class="false false"
        >
          <label
            class=" inline-block relative pl-5  cursor-pointer"
          >
            <input
              class="rounded focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
              type="checkbox"
              value="option2"
            />
            <span
              class="text-sm md:text-base"
            >
              オプション2
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
